### PR TITLE
Add GetterDict example

### LINKED ISSUE #1745: Document GetterDict for ORM mode properly properly
As per #1726 and #1689 we need to document `GetterDict` and how to use it properly.

### PR DESCRIPTION
**Pull request by nuno-andre**
Here is a very simple example of how to subclass `GetterDict` to process a class with no mapping protocol while dealing with the impedance mismatch between models (DOM's attributes and children vs OOP's properties).

I have added three optional fields to show "missing attribute", "missing children", and "missing children attribute".

closes #1745

### CODE REVIEW

**Comment by nuno-andre** [comment]
I've also documented the `_missing` sentinel behavior. But I'm not quite sure that's the intended way to raise missing field errors, as the `GetterDict` code seems to suggest that `KeyError` should do so (which would make a lot of sense IMO). However, this is not the case.

Could you please confirm this assumption?

**Comment by PrettyWood** [comment]
@nuno-andre Thanks a lot for this doc, it was needed!
And good remark for the `KeyError` thing, which would indeed make sense.
But for now better stick with what you did and maybe open a discussion to suggest a refacto?

**Comment by nuno-andre** [comment]
Of course, just wanted to confirm that I hadn't documented an unintended behavior or a regression.

**Comment by PrettyWood** [comment]
@nuno-andre can you add a change file please? :)

**Comment by nuno-andre** [comment]
Done! I've just realized I'd completely ignored the PR template. Sorry for that.

File: docs/usage/models.md [hunk 1]
```diff
@@ -157,14 +157,27 @@ Here a vanilla class is used to demonstrate the principle, but any ORM class cou
...
+Arbitrary classes are processed by *pydantic* using the `GetterDict` class (see
+[utils.py](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/utils.py)), which attempts to
 provide a dictionary-like interface to any class. You can customise how this works by setting your own
 sub-class of `GetterDict` as the value of `Config.getter_dict` (see [config](model_config.md)).
 
 You can also customise class validation using [root_validators](validators.md#root-validators) with `pre=True`. 
 In this case your validator function will be passed a `GetterDict` instance which you may copy and modify.
 
+The `GetterDict` instance will be called for each field with a sentinel as a fallback (if no other default
+value is set). Returning this sentinel (or `object()`) means that the field is missing. Any other value will
```
Comment (samuelcolvin) [inline docs/usage/models.md:172]:
```suggestion
value is set). Returning this sentinel means that the field is missing. Any other value will
```

Unless I'm missing something, you can't just return `object()` as that will give a new unique value which won't match `default`

**Review by samuelcolvin — COMMENTED** [review]
otherwise looks good.

**Comment by samuelcolvin** [comment]
please update, otherwise I'll accept that one tiny change some time soon and merge.

File: docs/usage/models.md (hunk 1 above)
Comment (nuno-andre) [reply docs/usage/models.md]: Yes, that's true. So strange, I remember checking it. I probably got mistaken on some test.

**Comment by nuno-andre** [comment]
I guess this CI error isn't related to the PR, right?

**Comment by samuelcolvin** [comment]
no, that was just a problem with that CI script.

**Pull request merged**